File: DAA_lab/lab1/1.c
#include <stdio.h>
#include <limits.h>

int main() {
    int n = 5;
    int arr[100] = {1, 2, 3, 4, 5};
    int smallest, ss, largest, sl;

    smallest = INT_MAX;
    ss = INT_MAX;
    largest = INT_MIN;
    sl = INT_MIN;

    // Find the smallest and largest elements
    for (int i = 0; i < n; i++) {
        if (arr[i] < smallest) {
            smallest = arr[i];
        }
        if (arr[i] > largest) {
            largest = arr[i];
        }
    }

    // Find the second smallest and second largest elements
    for (int i = 0; i < n; i++) {
        if (arr[i] > smallest && arr[i] < ss) {
            ss = arr[i];
        }
        if (arr[i] < largest && arr[i] > sl) {
            sl = arr[i];
        }
    }

    printf("Second smallest: %d\n", ss);
    printf("Second largest: %d\n", sl);

    return 0;
}

File: DAA_lab/lab1/1.py
import time
import random

def find_second_smallest_largest(arr):
    smallest = float('inf')
    largest = float('-inf')
    second_smallest = float('inf')
    second_largest = float('-inf')

    for num in arr:
        if num < smallest:
            second_smallest = smallest
            smallest = num
        elif num < second_smallest and num != smallest:
            second_smallest = num

        if num > largest:
            second_largest = largest
            largest = num
        elif num > second_largest and num != largest:
            second_largest = num

    return second_smallest, second_largest

def main():
    n = int(input("Enter the maximum power of 10 for array size: "))

    for power in range(1, n + 1):
        size = 10 ** power
        arr = [random.randint(0, 99) for _ in range(size)]

        start_time = time.time()
        
        second_smallest, second_largest = find_second_smallest_largest(arr)
        
        elapsed_time = time.time() - start_time

        print(f"Array size: {size}")
        print(f"Second smallest: {second_smallest}")
        print(f"Second largest: {second_largest}")
        print(f"Elapsed time: {elapsed_time:.6f} seconds\n")

if __name__ == "_main_":
    main()